src/transform/src/dataflow.rs

```diff
@@ -23,7 +23,7 @@ use std::collections::{HashMap, HashSet};
 /// This method is currently limited in scope to propagating filtering and
 /// projection information, though it could certainly generalize beyond.
 pub fn optimize_dataflow(dataflow: &mut DataflowDesc) {
-    optimize_dataflow_filters(dataflow);
```

Comment:
Mentioned on Slack; without this one of the queries in show.slt is misoptimized. Need to resolve this before merging.

Reply:
Fixed by #4362.